src/tools/synthetic_node.rs

```diff
@@ -437,18 +441,18 @@ impl Writing for InnerNode {
         _target: SocketAddr,
         payload: &[u8],
         buffer: &mut [u8],
-    ) -> Result<usize> {
+    ) -> io::Result<usize> {
         buffer[..payload.len()].copy_from_slice(payload);
         Ok(payload.len())
     }
 }
 
 #[async_trait::async_trait]
 impl Handshaking for InnerNode {
-    async fn perform_handshake(&self, mut conn: Connection) -> Result<Connection> {
+    async fn perform_handshake(&self, mut conn: Connection) -> io::Result<Connection> {
         match (self.handshake, !conn.side) {
             (Some(Handshake::Full), ConnectionSide::Initiator) => {
-                // Possible bug: running zebra node results in internal pea2pea panics:
+                // Possible bug: running zebra node io::Results in internal pea2pea panics:
```

Comment:
Heh - I think search and replace typo here?

Reply:
indeed, good catch!